### PR TITLE
tools/programmer: do not use programmer wrapper

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

While nice at first, this causes more issues than it's worth for a slightly more pretty output.

Hiding information from developers is not a good idea, especially when it comes to an often finicky programming step.

While in theory error output should be caught, there are enough cases where the programmer will get stuck in a loop or require user input that hiding programmer output by default is a bad idea.

### Issues/PRs references

alternative to #16370 and #16360

### CODE REVIEW

**Comment by fjmolinas** [comment]
I'm personally more inclined to ACK this as well, a wrapper script is nice, but it currently is not satisfying user needs, and although #16370 is an improvement, it's still not enough as for comments in https://github.com/RIOT-OS/RIOT/pull/16370#issuecomment-824809678.

I would be in favor of disabling by default and backporting this.

**Comment by fjmolinas** [comment]
@kaspar030 what do you think?

**Review by kaspar030 — APPROVED** [review]
ACK.

**Comment by kaspar030** [comment]
I agree. This is causing issues, which need to be at least mostly resolved before this should be default.

**Comment by kaspar030** [comment]
@benpicco could you provide a backport?

**Comment by kaspar030** [comment]
> @benpicco could you provide a backport?

Well actually I can :)

**Comment by kaspar030** [comment]
Backport provided in #16390

**Comment by aabadie** [comment]
Documentation was not updated accordingly:

https://github.com/RIOT-OS/RIOT/blob/31a6aa1c9c9b48c5d531f52ec2256dfb079a916b/makefiles/vars.inc.mk#L112

With this PR `USE_PROGRAMMER_WRAPPER_SCRIPT` is set to 0 by default.

File: makefiles/tools/programmer.inc.mk [hunk 1]
```diff
@@ -5,24 +5,8 @@ ifeq (0,$(PROGRAMMER_QUIET))
...
-ifneq (1,$(RIOT_CI_BUILD))
-  ifneq (,$(filter $(PROGRAMMER),$(PROGRAMMER_WRAPPER_BLACKLIST)))
-    USE_PROGRAMMER_WRAPPER_SCRIPT ?= 0
-  else
-    USE_PROGRAMMER_WRAPPER_SCRIPT ?= 1
-  endif
-else
-  USE_PROGRAMMER_WRAPPER_SCRIPT ?= 0
-endif
+# Don't use the programmer wrapper
```
Comment (aabadie) [inline makefiles/tools/programmer.inc.mk:8]: I would have added `by default` to this comment.

**Pull request merged**